cmake/traccc-ctest.sh.in

```diff
@@ -0,0 +1,17 @@
+#!@BASH_EXECUTABLE@
+#
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2024 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# Propagate errors.
+set -e
+set -o pipefail
+
+# Run every command through the time command. Recording the time, memory, etc.
+# used by each and every build command.
+command @TIME_EXECUTABLE@ --verbose \
+   --output=@CMAKE_CURRENT_BINARY_DIR@/traccc_build_performance.log --append \
+   @CMAKE_CTEST_COMMAND@ $*
```

Comment:
This work only with GNU time. The equivalent for BSD time would be:

```suggestion
command @TIME_EXECUTABLE@ -lao @CMAKE_CURRENT_BINARY_DIR@/traccc_build_performance.log @CMAKE_CTEST_COMMAND@ $*
```

Just to make it work on MacOS, too.

Reply:
I mean... sure. But we don't run any MacOS tests on this repository at the moment. 🤔 Does it actually build on it at the moment?

But even worse:

```
[bash][pcadp04]:~ > command time -lao ls
time: invalid option -- 'l'
Try 'time --help' for more information.
[bash][pcadp04]:~ >
```

So the command on RHEL9 doesn't actually like `-l`. That's a pretty big showstopper...

Reply:
I'd rather test for the OS to make this a bit more usable for our MacOS developer friends:

```bash
unamestr=$(uname)
if [ "$unamestr" = 'Linux' ]; then
   command @TIME_EXECUTABLE@ --verbose \
   --output=@CMAKE_CURRENT_BINARY_DIR@/traccc_build_performance.log --append \
   @CMAKE_CTEST_COMMAND@ $*
elif [ "$unamestr" = 'Darwin' ]; then
   command @TIME_EXECUTABLE@ -lao @CMAKE_CURRENT_BINARY_DIR@/traccc_build_performance.log @CMAKE_CTEST_COMMAND@ $*
fi
```

Reply:
😮‍💨 Let's support macOS then... But I didn't want to make that decision on every single build command.

In a perfect world CMake would actually test which of the `-l` or `-v` arguments are supported. But since this is not meant to be a production feature of the code, I didn't want to make it the most robust thing ever.

Though at this point, I might as well bite the bullet and go all out with testing during the CMake configuration which of the arguments works. Instead of hardcoding that `-l` would work on "Darwin". 🤔